src/map/lua/lua_item.cpp

```diff
@@ -370,6 +370,31 @@ void CLuaItem::setExData(sol::table const& newData)
     }
 }
 
+auto CLuaItem::getFishData() -> sol::table
+{
+    if (!fishingutils::IsFish(m_PLuaItem))
+    {
+        ShowError("ERROR: Attempted to retrieve fish data from non-fish item.");
+        return sol::lua_nil;
+    }
+
+    if (auto PFish = static_cast<CItemFish*>(m_PLuaItem))
+    {
+        sol::table table = lua.create_table();
+
+        table["length"] = PFish->GetLength();
+        table["weight"] = PFish->GetWeight();
+        table["ranked"] = PFish->IsRanked();
+
+        return table;
+    }
+    else
```

Comment:
This else is redundant, you'll get here if the if fails because of the return (we still want the logging and the lua_nil return)